backend-project/small_eod/generic/tests.py

```diff
@@ -31,11 +32,22 @@ def get_url(self, name, **kwargs):
             raise NotImplementedError("get_url must be overridden or basename defined")
         return reverse("{}-{}".format(self.basename, name), kwargs=kwargs)
 
-    def test_list_plain(self):
+    def test_dict_plain(self):
+        parsed_response_len = 4
```

Comment:
Testy nie przechodzą, więc nie wiem czy ten kod jest po prawny. Ale generalnie to potrzebujesz powyżej wspomnianego atrybutu klasy, a w tym miejscu zrobić "if", że jak ```self.paginated=True``` to przeanalizuj tak jakby to był słownik, a jeżeli nie to przeanalizuj jako listę. Warto zwrócić uwagę, że część będzie wspólna (sprawdzenie statusu odpowiedzi, wywołanie validated_item), więc warto to rozsądnie napisać, aby kod nie był powtórzony. Jak będzie, też będzie akceptowalnie.

Reply:
tzn chodzi mi w jaki sposób będziemy przekazywali do testów że akurat ten konkretny test jest z paginacją a inny bez, gdzie będe definiowane te zmnienne

Reply:
Jako atrybut klasy. Zobacz sobie jak ta klasa jest używana, bo już mamy kilka atrybutów klas. Przeczytaj także https://softwareengineering.stackexchange.com/questions/312339/are-python-mixins-an-anti-pattern https://coderbook.com/@marcus/deep-dive-into-python-mixins-and-multiple-inheritance/

Reply:
ok czaje po prostu przy dziedziczeniu z mixina będziemy nadpisywać zmienną klasową jeżeli chodzi o kodowanie raczej ogarniam bardziej mnie wizja zastanawiała

Reply:
Django CBV są w dużym stopniu w taki sposób napisane jako kompozycja Mixin, więc testy tego kodu w taki sposób naturalnie pasuje.